### PR TITLE
[DCJ-89][risk=no] Convert Spinner component to typescript

### PR DESCRIPTION
**Pull request by rushtong**
### Addresses
Addresses, but does not complete, work done for https://broadworkbench.atlassian.net/browse/DCJ-89

### Summary
This PR breaks out one of the TS conversions in https://github.com/DataBiosphere/duos-ui/pull/2512 to make that PR more manageable

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/Spinner.tsx [hunk 1]
```diff
@@ -2,11 +2,11 @@ import React from 'react';
...
-export const Spinner = (
-  <div style={{ textAlign: 'center', height: '44px', width: '180px' }}>
-    <img src={loadingIndicator} alt='spinner' />
-  </div>
-);
+export function Spinner() {
+  return <div style={{textAlign: 'center', height: '44px', width: '180px'}}>
+    <img src={loadingIndicator} alt='spinner'/>
+  </div>;
+}
```
Comment (rushtong) [inline src/components/Spinner.tsx:9]: This is a slightly different version of https://github.com/DataBiosphere/duos-ui/pull/2512/files#diff-2b35d22b68311c48b29ae44312e7ccb0450ef084cac25b5c9d72ed2f4ad2c8fd 
I found that using the full `(): JSX.Element` syntax was showing an ESLint error that didn't seem to make any sense.

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+declare module '*.svg'
```
Comment (rushtong) [inline src/index.d.ts]: This is necessary to get past an import error on the loading indicator on line 2 of `Spinner.tsx`

File: tsconfig.json [hunk 3]
```diff
@@ -15,7 +15,8 @@
     "target": "es2022",
     "useUnknownInCatchVariables": true,
     "paths": {
-      "@/*": ["src/*"]
+      "src/*": ["./src/*"],
+      "@/*": ["./*"]
```
Comment (rushtong) [inline tsconfig.json:19]: This is a pure copy/paste from https://github.com/DataBiosphere/duos-ui/pull/2512/files#diff-b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0

File: src/custom.d.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+declare module '*.svg' {
+    export const ReactComponent: React.FunctionComponent<
+        React.SVGAttributes<SVGElement>
+    >;
+
+    const src: string;
+    export default src;
+}
```
Comment (rushtong) [inline src/custom.d.ts:8]: This is a pure copy/paste from https://github.com/DataBiosphere/duos-ui/pull/2512/files#diff-5f2ca1b1541dc3023f81543689da349e59b97c708462dd8da4640b399362edc7

File: .eslintrc [hunk 5]
```diff
@@ -52,5 +52,13 @@
...
-  }
+  },
+  "overrides": [
+    {
+      "files": ["*.ts", "*.tsx"],
+      "rules": {
+        "no-undef": "off"
+      }
+    }
+  ]
```
Comment (rushtong) [inline .eslintrc:63]: This is a copy/paste from https://github.com/DataBiosphere/duos-ui/pull/2512/files#diff-e7e195d3d8fa5b82fa51ed952f3a179d429cbc46c0432ce1f7b357c54c162822

**Review by aarohinadkarni — APPROVED** [review]
looks good, thank you 👍

**Review by fboulnois — APPROVED** [review]
looks good, thank you!

**Pull request merged**